core-lib/Kernel.ns

```diff
@@ -410,8 +413,18 @@ class Kernel vmMirror: vmMirror = Object <: Value (
       ^ self substringFrom: argument to: argument
     )
 
+    public do: block = (
+      1 to: self length do: [ :i |
+        block value: (self charAt: i).
+      ]
+    )
+
     (* Printing *)
     public print = ( vmMirror printString: self )
+  ) : (
+    public from: charArray = (
```

Comment:
What's the error when `charArray` isn't what it is supposed to be? Invalid arg exception or something alike? would be good to document.